template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/StarterWeb-CSharp/Areas/IdentityService/IdentityServiceExtensions.cs

```diff
@@ -0,0 +1,84 @@
+using Microsoft.AspNetCore.Identity.Service.EntityFrameworkCore;
+using Microsoft.AspNetCore.Identity.Service.IntegratedWebClient;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Company.WebApplication1.Identity.Data;
+using Company.WebApplication1.Identity;
+using Company.WebApplication1.Identity.Models;
+using Company.WebApplication1.Identity.Services;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class IdentityServiceExtensions
+    {
+        public static IServiceCollection AddIdentityService(this IServiceCollection services, IConfiguration config)
+        {
+            // Add framework services.
+            services.AddDbContext<IdentityServiceDbContext>(options =>
+  #if (UseLocalDB)
+                options.UseSqlServer(config.GetConnectionString("DefaultConnection")));
+  #else
+                options.UseSqlite(config.GetConnectionString("DefaultConnection")));
+  #endif
+
+            services.AddIdentity<ApplicationUser, IdentityRole>()
+                .AddEntityFrameworkStores<IdentityServiceDbContext>()
+                .AddDefaultTokenProviders();
+
+            services.AddIdentityService<ApplicationUser, IdentityServiceApplication>(
+                options =>
+                {
+                    options.IdTokenOptions.ContextClaims.AddSingle("tfp", "policy");
+                    options.IdTokenOptions.ContextClaims.AddSingle("ver", "version");
+                    options.AccessTokenOptions.ContextClaims.AddSingle("tfp", "policy");
+                    options.AccessTokenOptions.ContextClaims.AddSingle("ver", "version");
+
+                    options.Issuer = config["Authentication:Identity:Issuer"];
+                })
+                .AddEntityFrameworkStores<IdentityServiceDbContext>()
+                .AddSigningCertificates(() =>
+                {
+                    var signingCertificates = config["Authentication:Identity:SigningCertificates"].Split(" ");
+                    return CertificateLoader.LoadAll(config)
+                        .Where(cert => signingCertificates.Contains(cert.Key))
+                        .Select(cert => cert.Value);
+                });
+
+            services.AddScoped<IAuthorizationResponseParameterProvider, ClientInfoProvider>();
+            services.AddScoped<ITokenResponseParameterProvider, ClientInfoProvider>();
+
+            // Add external authentication handlers below. To configure them please see https://go.microsoft.com/fwlink/?LinkID=532715
+
+            services.AddIntegratedWebClient(config.GetSection("Authentication:Identity"));
+
+            services.AddTransient<IEmailSender, AuthMessageSender>();
+            services.AddTransient<ISmsSender, AuthMessageSender>();
+
+            return services;
+        }
+
+        public static IServiceCollection AddIdentityServiceAuthentication(this IServiceCollection services)
```

Comment:
The naming here is a bit confusing, maybe add a comment to explain what the difference is between: 

AddIdentityServiceAuthentication and AddIdentityService